src/server/node_manager.c

```diff
@@ -7870,6 +7890,11 @@ update_job_node_rassn(job *pjob, attribute *pexech, enum batch_op op)
 				prdef = find_resc_def(svr_resc_def, pkvp[j].kv_keyw, svr_resc_size);
 				if (prdef == NULL)
 					return;
+	
+				if ((prdef->rs_flags & asgn) == 0) {
```

Comment:
This is the general fix to the crash seen at NASA due to presence of "aoe" in exec_vnode. This resource is not flagged to be set under resources_assigned. So the flag is check here to make sure it has a a flag of ANASSN or FNASSN and if not, just continue.